code/datums/components/holderloving.dm

```diff
@@ -0,0 +1,69 @@
+/** Holder Loving Component
+ *
+ * This component is assigned to an [/obj/item], and also keeps track of a [holder].
+ * The [parent] is 'bound' to [holder]. [parent] will be kept either directly
+ * inside [holder], or in the inventory of a [/mob] that is itself holding [holder].
+ *
+ * If [parent] is placed in a [loc] that is not [holder] or [holder].[loc]
+ * (if it's a mob), it is placed back inside [holder].
+ *
+ * This is intended for items that are a 'part' of another item.
+ *
+ * It can also delete [parent] when [holder] is deleted.
+ *
+ */
+/datum/component/holderloving
+	can_transfer = TRUE
+	/// Item that parent is bound to
+	var/obj/item/holder
+	/// If parent is deleted when the holder gets deleted
+	var/del_parent_with_holder = FALSE
+
+/datum/component/holderloving/Initialize(holder, del_parent_with_holder)
+	if(!isitem(parent) || !holder)
+		return COMPONENT_INCOMPATIBLE
+	src.holder = holder
+	src.del_parent_with_holder = (del_parent_with_holder || FALSE)
+
+/datum/component/holderloving/RegisterWithParent()
+	RegisterSignal(holder, COMSIG_MOVABLE_MOVED, .proc/check_my_loc)
+	RegisterSignal(holder, COMSIG_PARENT_QDELETING, .proc/holder_deleting)
+	RegisterSignal(parent, list(
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ITEM_EQUIPPED,
+		COMSIG_STORAGE_ENTERED,
+		COMSIG_STORAGE_EXITED,
+	), .proc/check_my_loc)
+
+/datum/component/holderloving/UnregisterFromParent()
+	UnregisterSignal(holder, list(COMSIG_MOVABLE_MOVED, COMSIG_PARENT_QDELETING))
+	UnregisterSignal(parent, list(
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ITEM_EQUIPPED,
+		COMSIG_STORAGE_ENTERED,
+		COMSIG_STORAGE_EXITED,
+	))
```

Comment:
Unregister in one list.

Reply:
holder and parent are different atoms, I don't see what you mean. Do you mean remove the indentation?

Reply:
Oh, nevermind.